samza-core/src/main/java/org/apache/samza/zk/ScheduleAfterDebounceTime.java

```diff
@@ -51,10 +50,23 @@
 
   public static final int DEBOUNCE_TIME_MS = 2000;
 
+  private final ScheduledTaskFailureCallback scheduledTaskFailureCallback;
+
   private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
       new ThreadFactoryBuilder().setNameFormat("debounce-thread-%d").setDaemon(true).build());
   private final Map<String, ScheduledFuture> futureHandles = new HashMap<>();
 
+  // Ideally, this should be only used for testing. But ZkBarrierForVersionUpgrades uses it. This needs to be fixed.
+  // TODO: Timer shouldn't be passed around the components. It should be associated with the JC or the caller of
+  // coordinationUtils.
+  public ScheduleAfterDebounceTime() {
```

Comment:
@sborya I couldn't easily add it to the schedule method because the callback should `stop` the JobCoordinator. So, if I were to replace all submissions to the scheduler, all those components should have access to `stop` method of `ZkJobCoordinator`. This doesn't feel right, esp. in ZkBarrierForVersionUpgrade, we will require a reference to the jobcoordinator. That's clunky and I don't want to add that now. 
I think this refactoring will require changes in SAMZA-1128 :(